### PR TITLE
Basic core dev docs

### PR DESCRIPTION
**Pull request by marc-casperlabs**
Adds the first of hopefully many documents detailing some questions that pop up during development for those hacking on the node proper.

It is built to be run with `mdbook` (run `make book` if you have it installed, then open `book/book/index.html`), however it can easily be viewed from GitHub. To give you an idea how to reach it, [check out the branch on GitHub](https://github.com/marc-casperlabs/casper-node/tree/d273-basic-docs) and then click the **[core dev book](https://github.com/marc-casperlabs/casper-node/blob/d273-basic-docs/book/src/SUMMARY.md)** link in the `README.md`. Or you can read directly from inside your IDE, of course.

### CODE REVIEW

File: book/src/architecture/announcement-howto.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
+# HOWTO: Adding an announcement
+
+A question that sometimes comes up is how to notify a component about a change that happened in another one. As components are decoupled they do not know about specific other components, so the way we achieve this is by having one component announce new information, and one or more react to said announcement, all wired up inside the reactor.
+
+Let's look at a concrete example (that we will never implement): We want the diagnostics console to send a message to all connected clients whenever a node starts or ceases to be come a validator. Our plan is to
```
Comment (rafal-ch) [inline book/src/architecture/announcement-howto.md:5]:
```suggestion
Let's look at a concrete example (that we will never implement): We want the diagnostics console to send a message to all connected clients whenever a node starts or ceases to become a validator. Our plan is to
```

File: book/src/architecture/announcement-howto.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+where
+    REv: From<ValidatorStatusChangedAnnouncement>,
+{
+    self.event_queue
+        .schedule(ValidatorStatusChangedAnnouncement { new_status }, QueueKind::Regular)
+        .await
+}
+```
+
+Now we can call `effect_builder.announce_validator_status_changed(..)` in another component and use it to return an effect that will cause the announcement to emitted.
```
Comment (rafal-ch) [inline book/src/architecture/announcement-howto.md:50]:
```suggestion
Now we can call `effect_builder.announce_validator_status_changed(..)` in another component and use it to return an effect that will cause the announcement to be emitted.
```

File: book/src/architecture/announcement-howto.md [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    ParticipatingEvent::ValidatorStatusChangedAnnouncement(_) => {
+        debug!("TODO: swallowed a validator status changed announcement");
+        Effects::new()
+    }
+  // ...
+```
+
+The necessary additions to the `description` and `Display::fmt` methods of `ParticipatingEvent` are not shown here.
+
+With these changes, the node will actually compile again and output a debug-level log messages when the announcement is made. We can now continue by implementing a consumer of this announcement.
```
Comment (rafal-ch) [inline book/src/architecture/announcement-howto.md:136]:
```suggestion
With these changes, the node will actually compile again and output debug-level log messages when the announcement is made. We can now continue by implementing a consumer of this announcement.
```

**Review by rafal-ch — APPROVED** [review]
LGTM :+1:

**Comment by Fraser999** [comment]
Closing for now.  @marc-casperlabs we'll relocate and refine these docs after further discussion.

**Pull request closed without merging**